src/module.c

```diff
@@ -4638,7 +4643,24 @@ void moduleHandleBlockedClients(void) {
             ctx.module = bc->module;
             ctx.client = bc->client;
             ctx.blocked_client = bc;
+            const ustime_t start = server.ustime;
             bc->reply_callback(&ctx,(void**)c->argv,c->argc);
+            const ustime_t reply_duration = ustime()-start;
+            c->duration += reply_duration;
+            /* Log the command into the Slow log if needed. */
+            if (!(c->cmd->flags & CMD_SKIP_SLOWLOG)) {
+                const char *latency_event = (c->cmd->flags & CMD_FAST) ?
+                                    "fast-command" : "command";
+                const ustime_t total_cmd_duration = c->duration + c->background_duration;
+                slowlogPushEntryIfNeeded(c,c->argv,c->argc,total_cmd_duration);
+                /* Log the reply duration either as fast-command or as command event. */
+                latencyAddSampleIfNeeded(latency_event,reply_duration/1000);
+                /* Log the background duration as background-command events. */
+                latencyAddSampleIfNeeded("background-command",c->background_duration/1000);
```

Comment:
i don't think we want the background duration in the latency monitor.
it is mainly used to find things that block the server (like a long `fsync` from the main thread)

Reply:
Solved in the last commit